pkgs/servers/haste-server/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, nixosTests
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, nodejs
+, pkgs
+}:
+
+stdenv.mkDerivation rec {
+  pname = "haste-server";
+  version = "3dcc43578b99dbafac35dece9d774ff2af39e8d0";
+
+  src = fetchFromGitHub {
+    owner = "toptal";
+    repo = "haste-server";
+    rev = version;
+    hash = "sha256-srSPRlG+gXSIwgVFLyfzRex97tCbV9FZXYpLD0KFRfw=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    makeWrapper
+  ];
+
+  installPhase =
+    let
+      nodeDependencies = ((import ./node-composition.nix {
+        inherit pkgs nodejs;
+        inherit (stdenv.hostPlatform) system;
+      }).nodeDependencies.override (old: {
+        # access to path '/nix/store/...-source' is forbidden in restricted mode
+        src = src;
+        dontNpmInstall = true;
+      }));
+    in
+    ''
+      runHook postInstall
+
+      mkdir -p $out/share
+      cp -ra . $out/share/haste-server
+      cp -ra ${nodeDependencies}/lib/node_modules $out/share/haste-server/node_modules
```

Comment:
Would it make a difference if we symlink this?

Reply:
We'd have two derivations then that would need to be pulled from cache. Not sure if we want this or instead just merge it into one.

Reply:
node_modules can be quiet big and doing ln will be way faster even if we need to pull two derivations.